### PR TITLE
add cumsum and cumprod ops to backend

### LINKED ISSUE #3792: add cumsum and cumprod ops
Hi

I made a PR #3791 to add these ops to the Keras backend directly. Do people think it is a good idea to add these ops?

### PR DESCRIPTION
**Pull request by kracwarlock**
This is a Keras 2 port of the PR #3791
Can also close #3792 if this is merged

### CODE REVIEW

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -1307,6 +1339,22 @@ def clip(x, min_value, max_value):
     return tf.clip_by_value(x, min_value, max_value)
 
 
+def clipnorm(x, max_value):
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: I don't think this one belongs. In any case, the name is questionable (inconsistent with TF), it does not have a Theano implementation, and no direct TF equivalent. Please remove.

Comment (kracwarlock) [inline keras/backend/tensorflow_backend.py]: Ah sorry that was an artifact from another PR. I have removed it now. You will have to squash and merge.

File: keras/backend/tensorflow_backend.py [hunk 2]
```diff
@@ -1062,6 +1062,38 @@ def prod(x, axis=None, keepdims=False):
     return tf.reduce_prod(x, reduction_indices=axis, keep_dims=keepdims)
 
 
+def cumsum(x, axis=None):
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: Set the default axis to be 0, for explicitness (same with the other one).

Comment (kracwarlock) [inline keras/backend/tensorflow_backend.py]: fixed
